lib/internal/cluster/master.js

```diff
@@ -104,8 +104,21 @@ function createWorkerProcess(id, env) {
   workerEnv.NODE_UNIQUE_ID = '' + id;
 
   if (execArgv.some((arg) => arg.match(debugArgRegex))) {
-    execArgv.push(`--inspect-port=${process.debugPort + debugPortOffset}`);
-    debugPortOffset++;
+    let inspectPort;
+    if ('inspectPort' in cluster.settings) {
+      if (typeof cluster.settings.inspectPort === 'function')
+        inspectPort = cluster.settings.inspectPort();
```

Comment:
I was going to open a follow up PR, but it could be done here.

Check if `cluster.settings.inspectPort` is a function. If so, call it and assign to `inspectPort`. If it's not a function, assign it directly to `inspectPort`. Then, call `require('internal/net').isValidPort(inspectPort)`, and throw an error if it's not valid.